react/features/overlay/components/OverlayContainer.js

```diff
@@ -0,0 +1,212 @@
+/* global APP */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import PageReloadOverlay from './PageReloadOverlay';
+import SuspendedOverlay from './SuspendedOverlay';
+import UserMediaPermissionsOverlay from './UserMediaPermissionsOverlay';
+
+/**
+ * Implements a React Component that will display the correct overlay when
+ * needed.
+ */
+class OverlayContainer extends Component {
+    /**
+     * OverlayContainer component's property types.
+     *
+     * @static
+     */
+    static propTypes = {
```

Comment:
There is something I don't like about these props.  They don't apply to the entire component, just to some of the sub-components.

Here is an alternative proposal: just create 2 props in mapStateToProps`: `overlayType` and `overlayOptions`.  Or, since we are taling about overlays, `type` and `options`.  Then in the render method you can create the right component based on the `type` prop and blindly pass the props in `options` using the spread operator. I think this would simplify the code quite a bit.